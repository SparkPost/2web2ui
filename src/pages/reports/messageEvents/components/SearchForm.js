import React from 'react';
import { Button, Inline, Panel } from 'src/components/matchbox';
import { FieldArray, reduxForm } from 'redux-form';
import { connect } from 'react-redux';
import { Form } from 'src/components/form';
import OGStyles from './SearchForm.module.scss';
import hibanaStyles from './SearchFormHibana.module.scss';
import { FORMS } from 'src/constants';
import EventTypeFilters from './EventTypeFilters';
import { getBooleanEventsObject, getSearchQueriesFromFilters } from '../helpers/transformData.js';
import { selectMessageEventListing } from 'src/selectors/eventListing';
import { getDocumentation, updateMessageEventsSearchOptions } from 'src/actions/messageEvents';
import SearchQuery from './SearchQuery';
import useHibanaOverride from 'src/hooks/useHibanaOverride';

const defaultFilters = [
  { key: 'recipient_domains' },
  { key: 'from_addresses' },
  { key: 'subjects' },
];

export function SearchForm(props) {
  const styles = useHibanaOverride(OGStyles, hibanaStyles);

  const { handleSubmit, handleApply, handleCancel, eventListing } = props;
  return (
    <Form onSubmit={handleSubmit(handleApply)} id="events-advanced-filter-form">
      <Panel.LEGACY title="Advanced Filters">
        <Panel.LEGACY.Section>
          <EventTypeFilters eventTypeDocs={eventListing} />
        </Panel.LEGACY.Section>
        <Panel.LEGACY.Section>
          <FieldArray component={SearchQuery} name="searchQuery" />
          <p>All filters accept comma-separated values.</p>
        </Panel.LEGACY.Section>
        <Panel.LEGACY.Section>
          <Inline>
            <Button variant="primary" submit>
              Apply Filters
            </Button>
            <Button className={styles.Cancel} variant="secondary" onClick={handleCancel}>
              Cancel
            </Button>
          </Inline>
        </Panel.LEGACY.Section>
      </Panel.LEGACY>
    </Form>
  );
}

const mapStateToProps = state => {
  const queryFilters = getSearchQueriesFromFilters(state.messageEvents.search);
  const initialFilters = queryFilters.length ? queryFilters : defaultFilters;

  return {
    initialValues: {
      searchQuery: [...initialFilters, {}],
      ...getBooleanEventsObject(state.messageEvents.search.events),
    },
    eventListing: selectMessageEventListing(state),
  };
};
export default connect(mapStateToProps, { updateMessageEventsSearchOptions, getDocumentation })(
  reduxForm({ form: FORMS.EVENTS_SEARCH, touchOnChange: true })(SearchForm),
);
