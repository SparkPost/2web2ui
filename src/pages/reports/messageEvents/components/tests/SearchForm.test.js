import React from 'react';
import { shallow } from 'enzyme';
import { SearchForm } from '../SearchForm';
jest.mock('src/hooks/useHibanaOverride', () => jest.fn(a => a));

describe('SearchForm', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      filterValues: {},
      onChange: jest.fn(),
      handleSubmit: jest.fn(a => a),
      handleCancel: jest.fn(),
      handleApply: jest.fn(),
    };
    wrapper = shallow(<SearchForm {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleApply when submitting', () => {
    const formValues = {};
    wrapper.find('Form').simulate('submit', formValues);
    expect(props.handleApply).toHaveBeenCalledWith(formValues);
  });

  it('should handle cancel when clicking the cancel button', () => {
    wrapper
      .find('Button')
      .last()
      .simulate('click');
    expect(props.handleCancel).toHaveBeenCalled();
  });
});
