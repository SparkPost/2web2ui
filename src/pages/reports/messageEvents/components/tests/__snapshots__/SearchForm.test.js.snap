// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should render correctly 1`] = `
<Form
  id="events-advanced-filter-form"
  onSubmit={[MockFunction]}
>
  <Panel.LEGACY
    title="Advanced Filters"
  >
    <Panel.LEGACY.Section>
      <EventTypeFilters />
    </Panel.LEGACY.Section>
    <Panel.LEGACY.Section>
      <FieldArray
        component={[Function]}
        name="searchQuery"
      />
      <p>
        All filters accept comma-separated values.
      </p>
    </Panel.LEGACY.Section>
    <Panel.LEGACY.Section>
      <Inline>
        <Button
          submit={true}
          variant="primary"
        >
          Apply Filters
        </Button>
        <Button
          className="Cancel"
          onClick={[MockFunction]}
          variant="secondary"
        >
          Cancel
        </Button>
      </Inline>
    </Panel.LEGACY.Section>
  </Panel.LEGACY>
</Form>
`;
